src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -154,12 +155,16 @@
 )
 
 # Bracket pairs (a set of tuples).
-# (name, startref, endref)
+# (name, startref, endref, persists)
+# NOTE: The `persists` value controls whether this type
+# of bracket is persisted during matching to speed up other
+# parts of the matching process. Round brackets are the most
+# common and match the largest areas and so are sufficient.
 ansi_dialect.sets("bracket_pairs").update(
     [
-        ("round", "StartBracketSegment", "EndBracketSegment"),
-        ("square", "StartSquareBracketSegment", "EndSquareBracketSegment"),
-        ("curly", "StartCurlyBracketSegment", "EndCurlyBracketSegment"),
+        ("round", "StartBracketSegment", "EndBracketSegment", True),
```

Comment:
Nitpick: The comment above is helpful. Maybe someday use `typing.NamedTuple` as well, to make the code more self documenting?

Reply:
Agreed. I'm going to leave it for now, but if it gets any more complicated we probably should.